### PR TITLE
docs: sd-tab docs [skip chromatic] 

### PR DESCRIPTION
**Pull request by smfonseca**
## Description:
This PR contains the rewrite for the following components:
- sd-tab
- sd-tab-panel
- sd-tab-group

## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/tab/tab.stories.ts [hunk 1]
```diff
@@ -33,129 +46,61 @@ export const Default = {
 };
 
 /**
- * Use the `active` attribute to toggle the active state. Styling to indicate the active state is applied in the `sd-tab-group` component.
+ * Use the `variant` attribute to display different styles.
```
Comment (coraliefeil) [inline packages/components/src/components/tab/tab.stories.ts:49]: Use the ”variant” attribute to change the appearance.

Comment (coraliefeil) [reply packages/components/src/components/tab/tab.stories.ts:49]: Default:  Can be used independently or within components, often for full-page content.
Container: Highlighted tab and content linked to a background container, typically for specific sections like sub-pages or teasers.

File: packages/components/src/components/tab/tab.stories.ts [hunk 2]
```diff
@@ -33,129 +46,61 @@ export const Default = {
...
-      axis: {
-        y: { type: 'attribute', name: 'disabled' }
-      },
-      args
-    });
-  }
+  name: 'Disabled',
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-tab disabled>Default</sd-tab>
```
Comment (coraliefeil) [inline packages/components/src/components/tab/tab.stories.ts:83]: <sd-tab disabled>Disabled</sd-tab>

File: packages/components/src/components/tab/tab.stories.ts [hunk 3]
```diff
@@ -33,129 +46,61 @@ export const Default = {
...
-      )}
-    `;
-  }
-};
 /**
- * Use the `left` slot to optionally include an element (eg. icon) positioned to the left of the label.
+ * Use the `left` slot include an element (eg. icon) positioned to the left of the label.
  */
+export const WithIcon = {
+  name: 'With Icon',
```
Comment (coraliefeil) [inline packages/components/src/components/tab/tab.stories.ts:92]: [Icon](name: 'Icon',)

Use the ”icon-left” slot to add system icons.

Comment (coraliefeil) [reply packages/components/src/components/tab/tab.stories.ts:92]: Bildschirmfoto 2024-09-03 um 14 47 07
Pls open up a bug issue - the gap between icon and label is missing

Comment (coraliefeil) [reply packages/components/src/components/tab/tab.stories.ts:92]: Pls remove container variant here

**Comment by coraliefeil** [comment]
[Figma is refined](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3060-29110&node-type=SECTION&t=Zgc7Tarofc9Ibrhm-11)

File: packages/components/src/components/tab/tab.stories.ts (hunk 2 above)
Comment (smfonseca) [reply packages/components/src/components/tab/tab.stories.ts:83]: Updated

File: packages/components/src/components/tab/tab.stories.ts (hunk 3 above)
Comment (smfonseca) [reply packages/components/src/components/tab/tab.stories.ts:92]: > [Icon](name: 'Icon',)
> 
> Use the ”icon-left” slot to add system icons.

In this component the slot for the icon is named `left`. If we want to update it to `icon-left` we need to open an issue.

> Bildschirmfoto 2024-09-03 um 14 47 07 Pls open up a bug issue - the gap between icon and label is missing

[Bug issue](https://github.com/solid-design-system/solid/issues/1345)

> Pls remove container variant here

Removed

File: packages/components/src/components/tab-panel/tab-panel.stories.ts [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
 import { withActions } from '@storybook/addon-actions/decorator';
 
 const { argTypes, parameters } = storybookDefaults('sd-tab-panel');
 const { overrideArgs } = storybookHelpers('sd-tab-panel');
 const { generateTemplate } = storybookTemplate('sd-tab-panel');
 
+/**
+ * Used inside `sd-tab-group` to display tabbed content.
+ *
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/src/components/tab-panel/tab-panel.stories.ts:12]: Updated

File: packages/components/src/components/tab/tab.stories.ts (hunk 3 above)
Comment (coraliefeil) [reply packages/components/src/components/tab/tab.stories.ts:91]: Alright we’ll keep "left"- Thanks

File: packages/components/src/components/tab-panel/tab-panel.stories.ts [hunk 5]
```diff
@@ -36,7 +53,23 @@ export default {
  */
 
 export const Default = {
+  name: 'Default',
```
Comment (auroraVasconcelos) [inline packages/components/src/components/tab-panel/tab-panel.stories.ts:56]: Same as above. No need to add name, only in screenshot tests

**Review by auroraVasconcelos — APPROVED** [review]

File: packages/components/src/components/tab-group/tab-group.stories.ts [hunk 6]
```diff
@@ -54,317 +64,35 @@ export const Default = {
...
+      </sd-tab-panel>
+      <sd-tab slot="nav" panel="tab-4" variant="default">Tab 4</sd-tab>
+      <sd-tab-panel name="tab-4">
+        <div>Inventore perspiciatis delectus nisi doloremque soluta inventore.</div>
+      </sd-tab-panel>
+      <sd-tab slot="nav" panel="tab-5" variant="default">Tab 5</sd-tab>
+      <sd-tab-panel name="tab-5">
+        <div>Provident nisi doloremque soluta inventore.</div>
+      </sd-tab-panel>
+    </sd-tab-group>
```
Comment (Vahid1919) [inline packages/components/src/components/tab-group/tab-group.stories.ts:96]: @smfonseca I would expect to see the "auto" version here as well. Only the manual tab-group is showcased in the story, which I feel won't be enough to showcase their differences.
image

**Review by Vahid1919 — CHANGES_REQUESTED** [review]

File: packages/components/src/components/tab-group/tab-group.stories.ts [hunk 7]
```diff
@@ -54,317 +64,35 @@ export const Default = {
...
-        const nextActivePanel = document.querySelector('sd-tab[panel="tab-5"]');
-
-        btn.addEventListener('click', () => {
-          sdTabGroup.setActiveTab(nextActivePanel);
-        });
-      </script>
-    `;
-  }
+export const Activation = {
+  name: 'Activation',
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/src/components/tab-group/tab-group.stories.ts]: Updated.

File: packages/components/src/components/tab-panel/tab-panel.stories.ts (hunk 5 above)
Comment (smfonseca) [reply packages/components/src/components/tab-panel/tab-panel.stories.ts]: Updated.

**Review by Vahid1919 — APPROVED** [review]

**Comment by coraliefeil** [comment]
Approved <3

**Pull request merged**